### PR TITLE
Fix inaccurate IK and gripper loose during transit

### PR DESCRIPTION
**Pull request by pointW**
1. Change the logic of function 'moveTo' to iteratively solve IK and approach to the goal position. Calling IK once can not get accurate solution. 
2. Keep a flag of if gripper is supposed to be closed or not. After sending position command and gripper is supposed to be closed, send another velocity command to gripper.
3. Remove the gripper position command while moving arm to some goal position
4. Fix bug that pre_rot is different from rot in picking

### CODE REVIEW

File: pybullet_toolkit/robots/ur5_rg2.py [hunk 1]
```diff
@@ -16,9 +16,9 @@ class UR5_RG2(object):
   '''
   def __init__(self):
     # Setup arm and gripper variables
-    self.max_forces = [150, 150, 150, 28, 28, 28, 100, 100]
-    self.gripper_close_force = [100] * 2
-    self.gripper_open_force = [100] * 2
+    self.max_forces = [150, 150, 150, 28, 28, 28, 500, 500]
+    self.gripper_close_force = [500] * 2
+    self.gripper_open_force = [500] * 2
```
Comment (ColinKohler) [inline pybullet_toolkit/robots/ur5_rg2.py]: There a reason you set the force to 500?

Comment (pointW) [reply pybullet_toolkit/robots/ur5_rg2.py]: not really?? just some random trials I guess... I can switch it back to 100

File: pybullet_toolkit/robots/ur5_rg2.py [hunk 2]
```diff
@@ -30,14 +30,20 @@ def reset(self):
...
     self.motor_indices = list()
+    self.arm_joint_names = list()
+    self.arm_joint_indices = list()
     for i in range (self.num_joints):
       joint_info = pb.getJointInfo(self.id, i)
       q_index = joint_info[3]
+      if i in range(1, 7):
+        self.arm_joint_names.append(str(joint_info[1]))
+        self.arm_joint_indices.append(i)
       if q_index > -1:
```
Comment (ColinKohler) [inline pybullet_toolkit/robots/ur5_rg2.py]: I think we can get rid of the motor indices/names and just replace them with arm and gripper indices/names.

File: pybullet_toolkit/robots/ur5_rg2.py [hunk 3]
```diff
@@ -48,7 +54,7 @@ def pick(self, pos, rot, offset, dynamic=True):
     pre_pos = copy.copy(pos)
     pre_pos[2] += offset
     # rot = pb.getQuaternionFromEuler([np.pi/2.,-np.pi,np.pi/2])
-    pre_rot = pb.getQuaternionFromEuler([0, np.pi, 0])
+    pre_rot = rot
```
Comment (ColinKohler) [inline pybullet_toolkit/robots/ur5_rg2.py]: Typically I don't care about the rotation when doing this positioning. Do you think this way is better?

File: pybullet_toolkit/robots/ur5_rg2.py [hunk 4]
```diff
@@ -80,43 +86,92 @@ def place(self, pos, rot, offset, dynamic=True):
...
+        self._setJointPoses(ik_solve)
+
+      ls = pb.getLinkState(self.id, self.end_effector_index)
+      new_pos = list(ls[4])
+      new_rot = list(ls[5])
+      close_enough = np.allclose(np.array(new_pos + new_rot), np.array(list(pos) + list(rot)), atol=threshold)
+      it += 1
+    pass
+
+    # # ik_solve = pb.calculateInverseKinematics(self.id, self.end_effector_index, pos, rot)
```
Comment (ColinKohler) [inline pybullet_toolkit/robots/ur5_rg2.py]: Might as well get rid of all these comments. If we need to look at this code later there is always the commit history.

File: pybullet_toolkit/robots/ur5_rg2.py (hunk 3 above)
Comment (pointW) [reply pybullet_toolkit/robots/ur5_rg2.py]: The previous one was actually wrong I guess. The pre grasp pose and grasp pose should have the same rotation

File: pybullet_toolkit/robots/ur5_rg2.py (hunk 4 above)
Comment (pointW) [reply pybullet_toolkit/robots/ur5_rg2.py]: yeah that make sense.

**Pull request merged**